pkg/dao/snapshots.go

```diff
@@ -71,12 +71,13 @@ func (sDao *snapshotDaoImpl) List(repoConfigUuid string, paginationData api.Pagi
 	order := convertSortByToSQL(paginationData.SortBy, sortMap, "created_at asc")
 
 	filteredDB := sDao.db.
-		Where("snapshots.repository_configuration_uuid = ?", UuidifyString(repoConfigUuid))
+		Model(&models.Snapshot{}).
+		Joins("JOIN repository_configurations ON repository_configuration_uuid = repository_configurations.uuid").
+		Where("repository_configurations.org_id = ?", orgID).
+		Where("repository_configuration_uuid = ?", UuidifyString(repoConfigUuid))
```

Comment:
I think we can leave it if there's not a performance difference, but I wonder, does it make sense to just filter by orgID [here](https://github.com/content-services/content-sources-backend/blob/main/pkg/dao/snapshots.go#L56-L58) instead of joining the two tables?

Reply:
My reasoning was there was a purpose-built error check to see if the repoUUID existed or no, so I still wanted that error check to occur. 
If you pass in the orgId above you would get an error when the desired outcome (in my mind) would simply be an empty list.

Reply:
I agree that if there are no matching snapshots, but the repo config is found, there should be an empty list. I think there's a couple ways you could do that. For example, you could query all of the ones associated to a particular repo config UUID, if there aren't any, return an empty list.

It probably makes sense to filter by org ID [here](https://github.com/content-services/content-sources-backend/blob/main/pkg/dao/snapshots.go#L56-L58) either way though, because the repo config should belong to the requester. If it doesn't, there's no need to do anything else.

Reply:
oh hmm, I am working on Red Hat Repositories at the moment, I'll be making some changes to this PR to support that and will see about adding the orgId.

Reply:
Okay, I added this, along with some functionality around ensuring redhat repositories will be returned.